include/kodlab_mjbots_sdk/joint_moteus.h

```diff
@@ -103,129 +90,219 @@ class JointMoteus: public JointBase
+
         /**
          * @brief Destroy the JointMoteus object, virtual to ensure proper 
          * destruction of derived classes
          */
         virtual ~JointMoteus(){}
 
         /**
-         * @brief Update the joint of the moteus. Converts rot/s to rad/s and saves mode
+         * @brief Determines whether the QueryCommand struct ignores position
+         * and velocity when requesting data from the moteus, potentially
+         * characteristic of an open-loop controller
          * 
-         * @param reply_pos position reported by moteus [rot]
-         * @param reply_vel velocity reported by moteus [rot/s]
-         * @param reply_torque torque measured by the moteus [Nm]
-         * @param mode      moteus mode
+         * @return bool 
          */
-        void UpdateMoteus(float reply_pos, float reply_vel, float reply_torque, ::mjbots::moteus::Mode mode ){
-            UpdateState( 2 * M_PI * reply_pos, 2 * M_PI * reply_vel, reply_torque);
-            mode_ = mode;
-        }
+        bool is_open_loop_query();
 
         /**
          * @brief Get the can id 
          * 
          * @return int 
          */
-        int get_can_id() const {return can_id_;}
+        int get_can_id() const;
 
         /**
          * @brief Get the can bus object
          * 
          * @return int 
          */
-        int get_can_bus() const {return can_bus_;}
+        int get_can_bus() const;
 
         /**
-         * @brief Get the mode reference 
+         * @brief Get the QueryCommand object, which contains resolutions
+         * 
+         * @return ::mjbots::moteus::QueryCommand
+         */
+        [[nodiscard]] ::mjbots::moteus::QueryCommand get_query_command() const;
+
+        /**
+         * @brief Get the mode reference
          * 
          * @return const ::mjbots::moteus::Mode& 
          */
-        const ::mjbots::moteus::Mode & get_mode_reference()   const {return mode_; }
+        const ::mjbots::moteus::Mode & get_mode_reference() const;
+
+        /**
+         * @brief Get the current operational mode of the servo
+         * 
+         * @return ::mjbots::moteus::Mode
+         */
+        [[nodiscard]] ::mjbots::moteus::Mode get_mode() const;
+
+        /**
+         * @brief Get the current position of the servo, measured in rotations
+         * of the output shaft. Overrides function in joint_base.h
+         * 
+         * @return float 
+         */
+        float get_position() const override;
+
+        /**
+         * @brief Get the current velocity of the servo, measured in Hz at the 
+         * output shaft. Overrides function in joint_base.h
+         * 
+         * @return float
+         */
+        float get_velocity() const override;
+
+        /**
+         * @brief Get the torque command. Overrides function in joint_base.h
+         * 
+         * @return float
+         */
+        float get_servo_torque() const override;
```

Comment:
See comment in cpp. get servo torque already defined in parent class, was throwing resolution warning when not part of query.

```suggestion
```